chain/client/src/tests/maintenance_windows.rs

```diff
@@ -0,0 +1,45 @@
+use crate::test_utils::setup_no_network;
+use actix::System;
+use futures::{future, FutureExt};
+use near_actix_test_utils::run_actix;
+use near_client_primitives::types::GetMaintenanceWindows;
+
+use near_o11y::testonly::init_test_logger;
+use near_o11y::WithSpanContextExt;
+
+/// get maintenance window from view client
+#[test]
+fn test_get_maintenance_windows() {
+    init_test_logger();
+    run_actix(async {
+        let (_, view_client) =
+            setup_no_network(vec!["test".parse().unwrap()], "other".parse().unwrap(), true, true);
+        let actor = view_client.send(
+            GetMaintenanceWindows { account_id: "test".parse().unwrap() }.with_span_context(),
+        );
+        let actor = actor.then(|res| {
+            assert_eq!(res.unwrap().unwrap(), vec![1..3, 3..5, 5..7, 7..9]);
```

Comment:
I'm wondering about the definition of the maintenance windows.
If you're a producer of block 5 - should the maintenance window start from 5 or from 6?

Reply:
because in the definition in docs, you say:
> The maintenance windows for a specific validator are future block height ranges in current epoch, in which the validator do not need produce block or chunk

Reply:
Maybe the documentation should just contain an example

Reply:
If you look at this example here: you return:
(1..3), (3..5), (5..7), (7..9)
So from what I understood from your explanation, this means that this validator DOESN'T produce blocks:
1,2 (coming from 1..3)    3,4 (coming from 3..5)    5,6 (coming from 5..7)    7,8 (coming from 7..9)

So this means that this validator doesn't produce any blocks?

Reply:
Ah, it is a bug and fixed.  I was also confused with `Rang`. :joy: